File: 167TwoSumII/submit.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        for(int i = 0; i < numbers.length; i++){
            int num = target - numbers[i];
            
            // 二分查找
            int l = i + 1;
            int r = numbers.length - 1; // 在[l, r]中间查找
            while(l <= r){
                int mid = l + (r-l)/2;
                if (numbers[mid] == num)
                    return new int[] {i+1, mid+1};
                else if (numbers[mid] < num)
                    l = mid + 1;
                else
                    r = mid - 1;
            }
        }
        return new int[] {};
    }
}

File: 167TwoSumII/submit.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        # 定义索引
        i = 0
        j = len(numbers) - 1
        
        while True:
            if (numbers[i] + numbers[j] > target):
                j -= 1
            elif (numbers[i] + numbers[j] < target):
                i += 1
            else:
                return [i+1, j+1]